### PR TITLE
contextual help

### PR DESCRIPTION
**Pull request by desnoes**
Contextual help for xreport and route edit forms
Placeholder for xreport and route edit forms

### CODE REVIEW

File: c2corg_ui/static/partials/contexthelp/xreport-place.html [hunk 1]
```diff
@@ -0,0 +1,4 @@
+<p><strong translate>Incident location</strong></p>
```
Comment (asaunier) [inline c2corg_ui/static/partials/contexthelp/xreport-place.html]: ``<h4 translate>``

File: c2corg_ui/static/partials/contexthelp/xreport-motivations.html [hunk 2]
```diff
@@ -0,0 +1,3 @@
+<p><strong translate>Motivations</strong></p>
```
Comment (asaunier) [inline c2corg_ui/static/partials/contexthelp/xreport-motivations.html]: ``<h4 translate>``

File: c2corg_ui/static/partials/contexthelp/xreport-modifications.html [hunk 3]
```diff
@@ -0,0 +1,3 @@
+<p><strong translate>Practical consequences</strong></p>
```
Comment (asaunier) [inline c2corg_ui/static/partials/contexthelp/xreport-modifications.html]: ``<h4 translate>``

File: c2corg_ui/static/partials/contexthelp/xreport-description.html [hunk 4]
```diff
@@ -0,0 +1,3 @@
+<p><strong translate>Details of the actual outing and the incident</strong></p>
```
Comment (asaunier) [inline c2corg_ui/static/partials/contexthelp/xreport-description.html]: ``<h4 translate>``

File: c2corg_ui/static/partials/contexthelp/xreport-conditions.html [hunk 5]
```diff
@@ -0,0 +1,3 @@
+<p><strong translate>Study of the conditions</strong></p>
```
Comment (asaunier) [inline c2corg_ui/static/partials/contexthelp/xreport-conditions.html]: ``<h4 translate>``

File: c2corg_ui/static/partials/contexthelp/route-equipement_rating.html [hunk 6]
```diff
@@ -0,0 +1,28 @@
+<p><strong translate>Protection</strong></p>
```
Comment (asaunier) [inline c2corg_ui/static/partials/contexthelp/route-equipement_rating.html]: Perhaps consider using ``<h4 translate>`` instead. Ditto for other "strong" paragraphs below.

File: c2corg_ui/templates/xreport/edit.html [hunk 7]
```diff
@@ -409,91 +442,132 @@ <h2 class="heading show-phone" translate>factors</h2>
...
               <textarea app-markdown-editor name="modifications" ng-model="xreport.locales[0].modifications" class="form-control"></textarea>
               <span class="glyphicon glyphicon-ok form-control-feedback" ng-show="xreport.locales[0].modifications"></span>
             </div>
 
+			
+			
             ## other_comments
             <div id="other_comments-group" class="form-group full">
-              <label translate>other_comments</label>
+              <label app-context-help
```
Comment (asaunier) [inline c2corg_ui/templates/xreport/edit.html]: Shouldn't you use a placeholder in the textarea instead?
By the way it seems the "?" icon is shown for no textarea label of the form: the label is clickable, the contextual help is displayed when clicking, but one hardly see this is possible to do this.

capture d ecran 2017-06-14 a 23 14 36

File: c2corg_ui/templates/xreport/edit.html [hunk 8]
```diff
@@ -409,91 +442,132 @@ <h2 class="heading show-phone" translate>factors</h2>
...
+              <label app-context-help 
+				context-help-title="{{'modifications' | translate}}" 
+				context-help-content-url="/static/partials/contexthelp/xreport-modifications.html"
+				translate>modifications</label>
               <textarea app-markdown-editor name="modifications" ng-model="xreport.locales[0].modifications" class="form-control"></textarea>
               <span class="glyphicon glyphicon-ok form-control-feedback" ng-show="xreport.locales[0].modifications"></span>
             </div>
 
+			
+
```
Comment (asaunier) [inline c2corg_ui/templates/xreport/edit.html]: Unnecessary blank lines added

File: c2corg_ui/static/partials/contexthelp/route-equipement_rating.html (hunk 6 above)
Comment (asaunier) [inline c2corg_ui/static/partials/contexthelp/route-equipement_rating.html]: But do you really need those subtitles? The window is already entitled using the ``context-help-title`` provided in the forms. Then adding subtitles in the partials seem redundant:

capture d ecran 2017-06-14 a 23 17 48

File: c2corg_ui/static/partials/contexthelp/xreport-increase_impact.html [hunk 9]
```diff
@@ -0,0 +1,11 @@
+<p><strong translate>Factors which accentuate the consequences</strong></p>
+<p translate>If relevant, describe the factors which might have made the situation worse :</p>
+<ul>
+	<li><span translate>remoteness</span></li>
```
Comment (asaunier) [inline c2corg_ui/static/partials/contexthelp/xreport-increase_impact.html]: tabs

File: c2corg_ui/static/partials/contexthelp/xreport-reduce_impact.html [hunk 10]
```diff
@@ -0,0 +1,9 @@
+<p><strong translate>Factors which diminished the consequences</strong></p>
+<p translate>If relevant, describe the factors which prevented the incident from being more serious :</p>
+<ul>
+	<li><span translate>ability to avoid problems</span></li>
```
Comment (asaunier) [inline c2corg_ui/static/partials/contexthelp/xreport-reduce_impact.html]: tabs

**Comment by desnoes** [comment]
@asaunier : thank you very much for the review and the advises. I will make the modifications.

**Comment by asaunier** [comment]
@desnoes thanks, I'll test this evening. But I can still see tabs in the diff: https://github.com/c2corg/v6_ui/pull/1693/files

What about the textareas in the xreport editing form? Some have placeholders but not context help, other have both, other have only context help. That's not very consistent. And there is no ? icon for textareas with context help, so it's quite useless.
capture du 2017-06-15 14-44-34

**Comment by desnoes** [comment]
@asaunier: is it possible to format text in placeolders as for contextual help with partials files ? This is the resason why i did not put placeorder in every text field.
I will check arnaud's example in order to find how he implements the ? for contextual help. I will do that this evening.

**Comment by asaunier** [comment]
> is it possible to format text in placeolders as for contextual help with partials files ?

Nope
But do you really need advanced formatting for textareas content? Those are only text, not predefined options.

**Comment by desnoes** [comment]
> But do you really need advanced formatting for textareas content? Those are only text, not predefined options.

Not really, i will reformat the texts.

**Comment by desnoes** [comment]
@asaunier : i commited a new version. Whenever possible i removed the link to the static file to restrict to a placeholder. Hope this will be the good one !

**Comment by asaunier** [comment]
@desnoes There was still errors with embedded quotes in the placeholders that made the xreport editing page crash. I have pushed a commit 8121adf fixing those problems and now it works.

Feel free to merge if it's OK for you. When clicking on the "Merge pull request" button, please choose "Squash and merge" so that the 4 commits are grouped into a single one.

Thanks 👍

File: c2corg_ui/static/partials/contexthelp/xreport-increase_impact.html [hunk 11]
```diff
@@ -0,0 +1,10 @@
+<p translate>If relevant, describe the factors which might have made the situation worse:</p>
+<ul>
+  <li><span translate>remoteness</span></li>
+  <li><span translate>level of commitment of the route</span></li>
+  <li><span translate>not being prepared for an incident</span></li>
+  <li><span translate>lack of experience of what to do in an incident/accident</span></li>
+  <li><span translate>lack of equipment</span></li>
+  <li><span translate>etc.</span></li>
```
Comment (asaunier) [inline c2corg_ui/static/partials/contexthelp/xreport-increase_impact.html]: Do you really need to translate ``etc.`?
If not simply remove ``<span translate>``.

File: c2corg_ui/static/partials/contexthelp/xreport-reduce_impact.html [hunk 12]
```diff
@@ -0,0 +1,8 @@
+<p translate>If relevant, describe the factors which prevented the incident from being more serious:</p>
+<ul>
+  <li><span translate>ability to avoid problems</span></li>
+  <li><span translate>expertise in rescue techniques</span></li>
+  <li><span translate>having foreseen plenty of spare time for the outing</span></li>
+  <li><span translate>having the appropriate equipment</span></li>
+  <li><span translate>etc.</span></li>
```
Comment (asaunier) [inline c2corg_ui/static/partials/contexthelp/xreport-reduce_impact.html]: Problaby not needed to translate ``etc.``

**Comment by asaunier** [comment]
@desnoes Don't forget that all those new interface strings should be translated in Transifex :P

**Comment by desnoes** [comment]
> When clicking on the "Merge pull request" button, please choose "Squash and merge" so that the 4 commits are grouped into a single one.

I did not see a Squash and merge option after clicking on Merge pull request.

**Comment by desnoes** [comment]
@asaunier: how do you handle quotes in the placeholders ?

**Comment by asaunier** [comment]
> I did not see a Squash and merge option after clicking on Merge pull request.

You should have clicked on the "arrow" button next to "Merge pull request":

capture d ecran 2017-06-16 a 08 37 15


> how do you handle quotes in the placeholders ?

You must escape them using a backslash: ``"{'Some text with \'quotes\' in it' | translate}"

**Comment by desnoes** [comment]
Sorry, i did not see the arrow button, i was focused on the 'Merge Pull Request'.

File: c2corg_ui/static/partials/contexthelp/route-equipement_rating.html [hunk 13]
```diff
@@ -0,0 +1,28 @@
...
+<p translate>This fields details the presence and quality of fixed anchors (bolts, pitons, spits) as protection points for rock or ice climbing.</p>
+
+<p><strong translate>P1 well bolted :</strong></p>
+<ul>
+  <li translate>Sport climbing bolted with 10 or 12mm expansion bolts, chemical bolts, etc. The route may not be equiped for abseil.</li>
+  <li translate>Required gear : <span translate>12 to 14 clippers, some of them lengthy, 1 to 2 slings to complete equipment in grade II pitches.</span></li>
```
Comment (asaunier) [inline c2corg_ui/static/partials/contexthelp/route-equipement_rating.html]: @desnoes You have forgotten to fix the problem above.

Comment (asaunier) [inline c2corg_ui/static/partials/contexthelp/route-equipement_rating.html]: In addition I see that a similar string is used in another partial => make sure the strings are exactly the same in order to have it only once in Transifex.

**Comment by desnoes** [comment]
shit!! i forgot this one. I will fix it the evening.

----- Mail original -----
De: "Alex Saunier" <notifications@github.com>
À: "c2corg/v6_ui" <v6_ui@noreply.github.com>
Cc: "desnoes" <mathieu.desnoes@free.fr>, "Mention" <mention@noreply.github.com>
Envoyé: Vendredi 16 Juin 2017 10:31:43
Objet: Re: [c2corg/v6_ui] contextual help (#1693)



@asaunier commented on this pull request. 


In c2corg_ui/static/partials/contexthelp/route-equipement_rating.html : > @@ -0,0 +1,28 @@
+<p><strong translate>Protection</strong></p>
+<p translate>This fields details the presence and quality of fixed anchors (bolts, pitons, spits) as protection points for rock or ice climbing.</p>
+
+<p><strong translate>P1 well bolted :</strong></p>
+<ul>
+  <li translate>Sport climbing bolted with 10 or 12mm expansion bolts, chemical bolts, etc. The route may not be equiped for abseil.</li>
+  <li translate>Required gear : <span translate>12 to 14 clippers, some of them lengthy, 1 to 2 slings to complete equipment in grade II pitches.</span></li> 

@desnoes You have forgotten to fix the problem above.

**Pull request merged**